tests/unit/regression/lars_unittest.cc

```diff
@@ -252,8 +252,23 @@ void lars_n_less_than_d_feature_test_templated()
 	CRegressionLabels* labels=new CRegressionLabels(lab);
 	SG_REF(labels);
 	CLeastAngleRegression* lars=new CLeastAngleRegression(false);
+	SG_REF(lars)
+
 	lars->set_labels(labels);
-	lars->train(features);
+
+	//Catch exceptions thrown when training, clean up
+	try
+	{
+		lars->train(features);
+	}
+	catch(...)
+	{
+		SG_UNREF(lars);
```

Comment:
it's quite painful to say this but this is a good place for goto :P
but then again you wanna throw the exception :)